mqtt-client/src/main/java/com/gojek/mqtt/connection/MqttConnection.kt

```diff
@@ -484,6 +484,16 @@ internal class MqttConnection(
                     timeTakenMillis = (clock.nanoTime() - subscribeStartTime).fromNanosToMillis()
                 )
                 runnableScheduler.scheduleMqttHandleExceptionRunnable(mqttException, true)
+            } catch (illegalArgumentException: IllegalArgumentException) {
```

Comment:
few suggestions:
- wrap exception inside MqttException. Use `MqttException(int reason, Throwable cause)`
reason can be REASON_CODE_INVALID_SUBSCRIPTION

Reply:
Done